wasmVirtualMachine.js

```diff
@@ -460,9 +575,53 @@ function build_module(byte_code) {
                 i++;
                 decode = Leb.decodeUint32(byte_code, i);
                 i = decode.nextIndex;
-                size = decode.value;
+                expected_end = i + decode.value;
+
+                // get array of table initializers
+                decode = Leb.decodeUint32(byte_code, i);
+                i = decode.nextIndex;
+                num_els = decode.value;
+
+                // get table initializer
+                for (let j = 0; j < num_els; j++) {
+                    // get table index 
+                    decode = Leb.decodeUint32(byte_code, i);
+                    i = decode.nextIndex;
+                    table_index = decode.value;
+
+                    // get offset
+                    let offset;
+                    let ret = parse_init_expression(byte_code, i);
+                    offset = ret.value;
+                    i = ret.nextIndex;
+
+                    // get elements (array of function indices)
+                    decode = Leb.decodeUint32(byte_code, i);
+                    i = decode.nextIndex;
+                    len = decode.value;
+
+                    // sanity check: 
+                    if (offset + len > tables[table_index].max_size) {
+                        console.log("table element indexing issue");
+                        return -1;
+                    }
+
+                    for (let k = 0; k < len; k++) {
+                        // get func index
+                        decode = Leb.decodeUint32(byte_code, i);
+                        i = decode.nextIndex;
+                        func_index = decode.value;
+
+                        // store func index at offset in element list
+                        tables[table_index].elements[offset + k] = func_index; 
```

Comment:
you initialize .elements to be an empty array. I'm not sure what happens if you index into a non-existing slot. You might need to grow the array first or set it's size

Reply:
Yeah I was confused at first, but in js you can index in an empty array and it will grow the array automatically with the first elements being empty items: here I assign the third element in an empty array and I get this [ <2 empty items>, 1 ]